web/src/stream_list.ts

```diff
@@ -822,13 +825,51 @@ export function set_event_handlers({
 }: {
     on_stream_click: (stream_id: number, trigger: string) => void;
 }): void {
-    $("#stream_filters").on("click", "li .subscription_block", (e) => {
+    $("#stream_filters").on(
+        "click",
+        "li .subscription_block .goto-interleaved-stream-view",
+        (e) => {
+            if (e.metaKey || e.ctrlKey) {
+                return;
+            }
+            const stream_id = stream_id_for_elt($(e.target).parents("li"));
+            on_stream_click(stream_id, "sidebar");
+
+            clear_and_hide_search();
+
+            e.preventDefault();
+            e.stopPropagation();
+        },
+    );
+
+    $("#stream_filters").on("click", "li .subscription_block .stream-name", (e) => {
         if (e.metaKey || e.ctrlKey) {
             return;
         }
+
         const stream_id = stream_id_for_elt($(e.target).parents("li"));
-        on_stream_click(stream_id, "sidebar");
+        const topics = stream_topic_history.get_recent_topic_names(stream_id);
+        let destination_url;
 
+        const top_unmuted_topic = topics.find(
+            (topic) => !user_topics.is_topic_muted(stream_id, topic),
+        );
+        const muted_topics = topics.find(
+            (topic) => !user_topics.is_topic_unmuted(stream_id, topic),
+        );
```

Comment:
So I think `muted_topics` here is wrong; can you talk about why you calculate it that way?

Reply:
Sorry for the confusion. I didn't realize that `muted_topics `still contains the **top topic**. Instead, it should be undefined when there are no muted topics."